src/Filter/AbstractDateFilter.php

```diff
@@ -83,12 +83,10 @@ final protected function filter(ProxyQueryInterface $query, string $field, Filte
 
         $value = $data->getValue();
 
-        if (!$value instanceof \DateTimeInterface) {
+        if (!$value instanceof \DateTime && !$value instanceof \DateTimeImmutable) {
```

Comment:
I think it's better to keep DateTimeInterface.

The error from phpstan detect a bug in DoctrineORM because we were using 
```
$endValue->add(new \DateInterval('P1D'));
```
with \DateTimeImmutable.

I would prefer to do `if(instanceof \DateTime) else {}`

Reply:
But here we're really only supporting `DateTime` and `DateTimeImmutable`, not really [`\DateTimeInterface` because it doesn't have the `add` method](https://www.php.net/manual/en/class.datetimeinterface.php). 

If another implementation of `DateTimeInterface` comes up, it won't be usable here.

Reply:
> If another implementation of `DateTimeInterface` comes up, it won't be usable here.

This is not possible, since the user land cannot implement it.
Static analysis already started to consider that `DateTimeInterface = Datetime || DatetImeImutable` and there is an issue to support the same in psalm.

Reply:
By `DateTimeInterface = Datetime || DatetImeImutable`, I mean

```
assert($value instanceof \DateTimeInterface);
assert(!$value instance \DateTime);
// Here $value is \DateTimeImmutable for phpstan (but still \DateTimeInterface for psalm currently)
```
This is pretty useful for `if(instanceof \DateTime) else {}` checks

Reply:
In DoctrineORMAdminBundle, 
```
$endValue = clone $value;
$endValue->add(new \DateInterval('P1D'));
```
were made for both DateTime and DateTimeImmutable.

It wasn't working for DateTimeImmutable, and it's useless to clone a DateTimeImmutable.
Here I would say that
```
elseif ($value instanceof \DateTime) {
                $endValue = clone $value;
                $endValue->add(new \DateInterval('P1D'));
            } else {
                /** @var \DateTimeImmutable $value */
                $endValue = $value->add(new \DateInterval('P1D'));
            }
```
reflects more the different logic of the DateTime and the DateTimeImutable.

Reply:
> > If another implementation of DateTimeInterface comes up, it won't be usable here.

> This is not possible, since the user land cannot implement it.

Yep, that's what I meant by comes up, if the core provides another one.

There are two different issues here, this PR meant to fix the check against `\DateTimeInterface` that to me it doesn't make any sense if we are using the `add` function which is not part of the interface (even if the interface means `Datetime || DatetimeImmutable` for SA tools).

The other issue about avoiding the `clone` call on the `DateTimeImmutable`, I don't see that as important, but if you think it is, it could be fixed.